config/rector/sets/authentication40.php

```diff
@@ -0,0 +1,30 @@
+<?php
+declare(strict_types=1);
+
+use Cake\Upgrade\Rector\Rector\MethodCall\RemoveMethodCallRector;
+use Cake\Upgrade\Rector\ValueObject\RemoveMethodCall;
+use Rector\Config\RectorConfig;
+use Rector\Renaming\Rector\Name\RenameClassRector;
+
+/**
+ * @see https://github.com/cakephp/authentication/blob/4.x/docs/en/upgrade-3-to-4.rst
+ */
+return static function (RectorConfig $rectorConfig): void {
+    // URL checker class renames
+    // Note: Order matters - StringUrlChecker rename must come first to avoid
+    // CakeRouterUrlChecker -> DefaultUrlChecker -> StringUrlChecker chain
+    $rectorConfig->ruleWithConfiguration(RenameClassRector::class, [
+        // Old DefaultUrlChecker renamed to StringUrlChecker
+        'Authentication\UrlChecker\DefaultUrlChecker' => 'Authentication\UrlChecker\StringUrlChecker',
+        // CakeRouterUrlChecker renamed to DefaultUrlChecker
+        'Authentication\UrlChecker\CakeRouterUrlChecker' => 'Authentication\UrlChecker\DefaultUrlChecker',
```

Comment:
If you run this more than once will all the UrlChecker references end up as `StringUrlChecker`?